### ISSUE TITLE
inline set error as warning

### ISSUE THREAD

**Issue by dsernst** [eslint/eslint#4201]
I'm looking for something similar to [`/* eslint-disable rule-foobar */`](https://github.com/eslint/eslint/pull/1840), but instead of setting the rule to 0, I'd like to set it to 1 (a warning).

I tried this:

```js
// Lots of code up here where I want `no-undef` to be set to error

/* eslint no-undef:1 */
var hits = data.hits.map(parseCompanySearchResult);
/* eslint no-undef:2 */

// More code down here where I want `no-undef` to be set to error
```

But it changes the entire file to whichever setting is last called, instead of splitting into sections of code.

Is there any way to accomplish this?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by dsernst** [comment]
For context, I'm adding `eslint` to a very large codebase, and trying to deal with the thousands of errors this has exposed. Linting is also getting added to continuous integration, so I can't move forward until fixing these errors.

In this particular case, the original author is not around any more. The functionality is important, so I don't want to make too many rash assumptions about what they meant.

So ideally I'd set this particular line to a warning for now to get past CI, but keep it marked for us to come back to when we have more time.

**Comment by dsernst** [comment]
Something like this would work:

```js
var hits = data.hits.map(parseCompanySearchResult); /* eslint-warn no-undef */
```

Or even a manual warning:

```js
var hits = data.hits.map(parseCompanySearchResult); /* eslint-manual-warn Come back and fix this line! */
```

**Comment by ilyavolodin** [comment]
Unfortunately all inline configuration (i.e. `/* eslint */` comments) is hoisted to the top of the file. The only suggestion that I have is to change your code this way:
```js
/* eslint "no-warning-comments": [1, { "terms": ["todo", "fixme"] }] */
/* eslint-disable no-undef */
 var hits = data.hits.map(parseCompanySearchResult); // todo: clean this up for no-undef rule
/* eslint-enable no-undef */
```

This way you get a warning for your todo comment

**Comment by platinumazure** [comment]
Could a CLI option work? More specifically, the idea is you would continue to use `eslint-disable`, but instead of outright ignoring the error, it would be logged as a warning.

The reason I'm suggesting a CLI option is because I think whether rule violations should be ignored or warned seems to be a decision that should probably apply to the whole repository, rather than on a case by case basis. If it is possible to both ignore and warn on specific violations, then cleaning up the warnings could result in a false sense of security since the remaining issues aren't reported. In addition, this would play very well with the `--max-warnings` CLI option, because it would allow you to fail a CI build if there are simply too many of these problems for your liking (something you could never catch with some ignores and some warnings).

**Comment by platinumazure** [comment]
Oh hell, I misread the issue and solved an XY problem. I like my idea anyway, though, and will propose it as a separate issue.

**Comment by nzakas** [comment]
No, we track severity by file, not by location. Attempting to have different severities for the same rule within a file would be a significant rewrite of a lot of code. I don't think it's with the complexity.

**Comment by dsernst** [comment]
Very clever @ilyavolodin . Thank you
